src/platform/forms-system/src/js/actions.js

```diff
@@ -210,41 +210,48 @@ export function uploadFile(
   // This item should have been set in any previous API calls
   const csrfTokenStored = localStorage.getItem('csrfToken');
   return (dispatch, getState) => {
-    if (file.size > uiOptions.maxSize) {
+    // we limit file types, but it’s not respected on mobile and desktop
+    // users can bypass it without much effort
+    if (
+      !uiOptions.fileTypes.some(fileType =>
+        file.name.toLowerCase().endsWith(fileType.toLowerCase()),
+      )
+    ) {
```

Comment:
Why is this check being moved so that it happens first?

Reply:
In case PDFs aren't an allowed type, I thought it'd be better not to show a file size error for an unsupported type.

Reply:
You're probably right, but I think it would be better to leave this out of the PR and keep it scoped to adding the max PDF option. I feel like when the order of `if`s and `return`s are changed around, something could easily go wrong :sweat_smile: 

Alternatively, does the current testing in place ensure that even if the order changes, the outcome (messaging) will remain the same?

Reply:
Ok, I've reverted that change.